src/main/java/org/jabref/gui/actions/ExportLinkedFilesAction.java

```diff
@@ -0,0 +1,169 @@
+                .withInitialDirectory(Paths.get(Globals.prefs.get(JabRefPreferences.EXPORT_WORKING_DIRECTORY)))
+                .build();
+        entries = JabRefGUI.getMainFrame().getCurrentBasePanel().getSelectedEntries();
+        exportPath = DefaultTaskExecutor
+                .runInJavaFXThread(() -> ds.showDirectorySelectionDialog(dirDialogConfiguration));
+
+        exportPath.ifPresent(path -> {
+            databaseContext = JabRefGUI.getMainFrame().getCurrentBasePanel().getDatabaseContext();
+            totalFilesCount = entries.stream().flatMap(entry -> entry.getFiles().stream()).count();
+
+            Service<Void> exportService = new ExportService();
+            startServiceAndshowProgessDialog(exportService);
+        });
+    }
+
+    private <V> void startServiceAndshowProgessDialog(Service<V> service) {
+        DefaultTaskExecutor.runInJavaFXThread(() -> {
+            service.start();
+
+            ProgressDialog progressDialog = new ProgressDialog(service);
+            progressDialog.setOnCloseRequest(evt -> service.cancel());
+            DialogPane dialogPane = progressDialog.getDialogPane();
+            dialogPane.getButtonTypes().add(ButtonType.CANCEL);
+            Button cancelButton = (Button) dialogPane.lookupButton(ButtonType.CANCEL);
+            cancelButton.setOnAction(evt -> {
+                service.cancel();
+                progressDialog.close();
+            });
+            progressDialog.showAndWait();
+        });
+    }
+
+    private class ExportService extends Service<Void> {
+
+        private static final String LOGFILE = "exportLog.log";
+        private final String localizedSucessMessage = Localization.lang("Copied file successfully");
+        private final String localizedErrorMessage = Localization.lang("Could not copy file") + ": " + Localization.lang("File exists");
+
+        @Override
+        protected Task<Void> createTask() {
+            return new Task<Void>() {
+
+                int totalFilesCounter;
+                int numberSucessful;
+                int numberError;
+                Optional<Path> newPath;
+
+                @Override
+                protected Void call()
+                        throws InterruptedException, IOException {
+                    updateMessage(Localization.lang("Copying files..."));
+                    updateProgress(0, totalFilesCount);
+
+                    try (BufferedWriter bw = Files.newBufferedWriter(exportPath.get().resolve(LOGFILE), StandardCharsets.UTF_8)) {
+
+                        for (int i = 0; i < entries.size(); i++) {
+
+                            List<LinkedFile> files = entries.get(i).getFiles();
+
+                            for (int j = 0; j < files.size(); j++) {
+                                updateMessage(Localization.lang("Copying file %0 of BibEntry %1", Integer.toString(j + 1), Integer.toString(i + 1)));
```

Comment:
I think we never used `BibEntry` before in the user interface (it is the class name...). What about only "entry" or "item"?